python/lsst/pipe/tasks/subtractBrightStars.py

```diff
@@ -259,95 +455,339 @@ def scaleModel(self, model, star, inPlace=True, nb90Rots=0):
+                minValidAnnulusFraction=self.warper.config.minValidAnnulusFraction,
+            )
+            for j, (warp, transform) in enumerate(
+                zip(self.warpOutputs.warpedStars, self.warpOutputs.warpTransforms)
+            )
+        ]
+        return brightStarList
+
+    def initAnnulusImage(self):
+        """Initialize an annulus image of the given star.
+
+        Returns
+        -------
+        annulusImage : `~lsst.afw.image.MaskedImageF`
+            The initialized annulus image.
+        """
+        maskPlaneDict = self.model.mask.getMaskPlaneDict()
+        annulusImage = MaskedImageF(self.modelStampSize, planeDict=maskPlaneDict)
+        annulusImage.mask.array[:] = 2 ** maskPlaneDict["NO_DATA"]
+        return annulusImage
+
+    def createAnnulus(self, brightStarStamp):
+        """Create a circular annulus around the given star.
+
+        The circular annulus is set based on the inner and outer optimal radii.
+        These radii describe to the annulus from where the flux of the star is found.
+        The aim is to create the same annulus for the PSF model, eventually
+        measuring the model flux around that annulus.
+        An optimal radius usually differes from the radius where the PSF model
+        is normalized.
+
+        Parameters
+        ----------
+        brightStarStamp :
+            `~lsst.meas.algorithms.brightStarStamps.BrightStarStamp`
+            A stamp of a bright star to be subtracted.
+
+        Returns
+        -------
+        annulus : `~lsst.afw.image.MaskedImageF`
+            An annulus of the given star.
+        """
+        # Create SpanSet of annulus.
+        outerCircle = SpanSet.fromShape(
+            brightStarStamp.optimalOuterRadius, Stencil.CIRCLE, offset=self.warper.modelCenter
+        )
+        innerCircle = SpanSet.fromShape(
+            brightStarStamp.optimalInnerRadius, Stencil.CIRCLE, offset=self.warper.modelCenter
+        )
+        annulus = outerCircle.intersectNot(innerCircle)
+        return annulus
+
+    def applyStatsControl(self, annulusImage):
+        """Apply statistics control to the PSF annulus image.
+
+        Parameters
+        ----------
+        annulusImage : `~lsst.afw.image.MaskedImageF`
+            An image containing an annulus of the given model.
+
+        Returns
+        -------
+        annularFlux: float
+            The annular flux of the PSF model at the radius where the flux of
+            the given star is determined.
+        """
+        andMask = reduce(
+            ior, (annulusImage.mask.getPlaneBitMask(bm) for bm in self.warper.config.badMaskPlanes)
+        )
+        self.missedStatsControl.setAndMask(andMask)
+        annulusStat = makeStatistics(annulusImage, self.missedStatsFlag, self.missedStatsControl)
+        return annulusStat.getValue()
+
+    def findPsfAnnularFlux(self, brightStarStamp, maskedModel):
+        """Find the annular flux of the PSF model at the radius where the flux
```

Comment:
I understand annular flux but am still confused by what "the radius where the flux of the given star is determined" means. Is it the radius of an aperture, and if so of what shape? Or a radius where the mean surface brightness is some specific value?